service/matching/taskQueueManager.go

```diff
@@ -494,27 +491,20 @@ func (c *taskQueueManagerImpl) GetUserData(ctx context.Context) (*persistencespb
 //nolint:revive // control coupling
 func (c *taskQueueManagerImpl) UpdateUserData(ctx context.Context, options UserDataUpdateOptions, updateFn UserDataUpdateFunc) error {
 	newData, err := c.db.UpdateUserData(ctx, updateFn, options.TaskQueueLimitPerBuildId)
-	c.signalIfFatal(err)
 	if err != nil {
 		return err
 	}
-	if options.Replicate && c.namespaceReplicationQueue != nil {
-		err = c.namespaceReplicationQueue.Publish(ctx, &replicationspb.ReplicationTask{
-			TaskType: enumsspb.REPLICATION_TASK_TYPE_TASK_QUEUE_USER_DATA,
-			Attributes: &replicationspb.ReplicationTask_TaskQueueUserDataAttributes{
-				TaskQueueUserDataAttributes: &replicationspb.TaskQueueUserDataAttributes{
-					NamespaceId:   c.taskQueueID.namespaceID.String(),
-					TaskQueueName: c.taskQueueID.BaseNameString(),
-					UserData:      newData.GetData(),
-				},
-			},
-		})
-		if err != nil {
-			c.logger.Error("Failed to publish a replication task after updating task queue user data", tag.Error(err))
-			return serviceerror.NewUnavailable("storing task queue user data succeeded but publishing to the namespace replication queue failed, please try again")
-		}
+	c.signalIfFatal(err)
+	_, err = c.matchingClient.ReplicateTaskQueueUserData(ctx, &matchingservice.ReplicateTaskQueueUserDataRequest{
+		NamespaceId: c.db.namespaceID.String(),
+		TaskQueue:   c.taskQueueID.BaseNameString(),
+		UserData:    newData.GetData(),
+	})
+	if err != nil {
+		c.logger.Error("Failed to publish a replication task after updating task queue user data", tag.Error(err))
+		return serviceerror.NewUnavailable("storing task queue user data succeeded but publishing to the namespace replication queue failed, please try again")
```

Comment:
Not quite direct publishing to queue any more, maybe slight rewording

Reply:
This LGTM TBH.